### ISSUE TITLE
Could not connect to the endpoint URL: "https://coiled-runtime-ci.s3.None.amazonaws.com/"

### ISSUE THREAD

**Issue by crusaderky** [coiled/benchmarks#1000]
Tests have started failing a few hours ago, with the message

```
botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "https://coiled-runtime-ci.s3.None.amazonaws.com/"
```
e.g. https://github.com/coiled/benchmarks/actions/runs/6253354999/job/16978575620?pr=999

@ntabris could you look into this?

**Comment by crusaderky** [comment]
CC @phofl I believe you experienced the same in your own tests

**Comment by ntabris** [comment]
> coiled-runtime-ci.s3.none.amazonaws.com

I could be wrong but I don't think there's supposed to be a `none` in that url. Any idea where that's coming from?

**Comment by crusaderky** [comment]
Our code is asking boto3 to write a file to
`s3://coiled-runtime-ci/test-scratch/cluster-dumps/{client.cluster.name}/<unique test name>`
where `client.cluster` is an instance of `coiled.Cluster`. Its `name` attribute is the only thing that could turn to None in that URI.

**Comment by crusaderky** [comment]
This is happening again. The result is that the whole CI is K.O.

```
INFO     coiled-runtime:conftest.py:645 Cluster state dump can be found at: s3://coiled-runtime-ci/test-scratch/cluster_dumps/test_repeated_merge_spill-63fbf224/stability.test_deadlock.py.test_repeated_merge_spill.msgpack.gz

E   botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "https://coiled-runtime-ci.s3.none.amazonaws.com/test-scratch/cluster_dumps/test_repeated_merge_spill-63fbf224/stability.test_deadlock.py.test_repeated_merge_spill.msgpack.gz"
```

**Comment by ntabris** [comment]
Some code is trying to build a regional S3 endpoint, which has the form `https://<bucket>.<region_name>.amazonaws.com`, and the region name is `None`.

I have no idea where that's happening, or where `None` is happening. If anyone has a consistent reproducer, then it would be possible to dig in, but without this I'm not sure how to debug.

Anyone know how to get this consistently? (Also, does it only happen in CI?)

We are setting `DEFAULT_REGION` on the cluster. That's shipped from local env var (local in this case means on the CI runner). It might be worth checking that this is set to a valid value.

As a workaround, it might work to explicit set region in your code that uses s3.

**Comment by crusaderky** [comment]
It turns out that all tests that read or write data on s3://coiled-runtime-ci/ are broken. Disabling cluster dumps only fixes the subset of tests that doesn't perform any S3 I/O - which is to say, not many.

**Comment by dchudz** [comment]
@crusaderky We'd like to help but it's hard without an example (either a reproducer or a fails-to-reproduce...) we can run outside your CI. 

If you can give us that, one of us will happily dig in. Is that reasonable?

**Comment by crusaderky** [comment]
@dchudz yes just open a blank PR.

I'm failing to reproduce just with a notebook, even if I use the account `dask-benchmarks`.
Notably, though, on my notebook I don't have the following environment variables set:

```yaml
DASK_COILED__TOKEN: ${{ secrets.COILED_BENCHMARK_BOT_TOKEN }}
AWS_ACCESS_KEY_ID: ${{ secrets.RUNTIME_CI_BOT_AWS_ACCESS_KEY_ID }}
AWS_SECRET_ACCESS_KEY: ${{ secrets.RUNTIME_CI_BOT_AWS_SECRET_ACCESS_KEY }}
```

**Comment by dchudz** [comment]
@crusaderky is there anything special I need to do to run locally? 

it's OK if it doesn't repro.

**Comment by dchudz** [comment]
@crusaderky what do I need to do to print output I can see in the CI run? 

I added some prints ("boto default region_name is") in https://github.com/coiled/benchmarks/pull/1029. The test they're in is running (and failing) but I don't see my output.

https://github.com/coiled/benchmarks/actions/runs/6341971500/job/17226775013?pr=1029

**Comment by crusaderky** [comment]
> @crusaderky is there anything special I need to do to run locally?
> 
> it's OK if it doesn't repro.

```python
import coiled
import distributed

cluster = coiled.Cluster(
    # kwargs from cluster_kwargs.yaml
    package_sync=True,
    wait_for_workers=True,
    scheduler_vm_types=["m6i.large"],
    backend_options={
      "spot": True,
      "spot_on_demand_fallback": True,
      "multizone": True,
    }
    n_workers=1,
    worker_vm_types=["m6i.large"],
    account="dask-benchmarks",
)

client = distributed.Client(cluster)
# The workers will use fsspec to write to s3
client.dump_cluster_state("s3://coiled-runtime-ci/test-scratch/foo-DELETEME.msgpack.gz")
```

**Comment by dchudz** [comment]
@crusaderky I'm struggling b.c it seems like after the first couple changes, my tests stop running on PRs

https://github.com/coiled/benchmarks/pull/1032

any advice?

**Comment by crusaderky** [comment]
Found a fix!
If I export the env variable

AWS_DEFAULT_REGION: us-east-2

before I invoke pytest, the problem disappears. That hack was already there elsewhere in the CI script to invoke the aws cli; no idea why boto3 worked until now.

**Comment by crusaderky** [comment]
> @crusaderky I'm struggling b.c it seems like after the first couple changes, my tests stop running on PRs
> 
> #1032
> 
> any advice?

it's still busy running the previous iteration of your tests; there are limited slots. You have to manually cancel the runs you don't care about in the Actions tab

**Comment by crusaderky** [comment]
Closed by #1029

**Comment by ntabris** [comment]
Well, turns out that this is my fault—sorry!

Specifically, there was an unintended change in coiled client (made by me) such that if default boto session region was `None` locally, it then shipped "None" to the cluster. In most cases there is a default session, but that wasn't the case for the way creds were set in benchmarks CI.

I expect we'll get fix out in next client release.
